common/etc/nginx/include/s3gateway.js

```diff
@@ -988,8 +988,9 @@ async function fetchCredentials(r) {
     }
 
     if (current) {
-        // AWS returns Unix timestamps in seconds, but in Date constructor we should provide timestamp in milliseconds
-        const exp = new Date(current.expiration * 1000).getTime() - maxValidityOffsetMs;
+        // If AWS returns a Unix timestamp it will be in seconds, but in Date constructor we should provide timestamp in milliseconds
```

Comment:
Thank you for your contribution. This looks good, but I would like you to write a longer comment that explains the reason for the change in more detail. Also please provide links to the relevant AWS documentation. I hope to merge this soon.

Reply:
Hi Matthew, I was thinking that you add a comment that is basically what you wrote above, but a bit more succinct.

Reply:
Ah of course! Sorry - I've added an explanatory comment.

I still couldn't find decent documentation for the ECS/Fargate metadata endpoints. But I was able to find better documentation for the EC2 and STS endpoints that are also supported. It appears that neither of these return expiration times as unix timestamps, they both return RFC 3339 strings instead:

- https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#instance-metadata-security-credentials
- https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html

So it may be that the multiplication by 1000 is completely redundant. But I'm not proposing to remove it in this MR as there may be some situation I'm not aware of where it does come into play.